server/sources/ElectionManager/ServerElectionManager.cpp

```diff
@@ -0,0 +1,228 @@
+//
+
+#include "ServerElectionManager.h"
+#include <iostream>
+#include <netinet/in.h>
+#include <pthread.h>
+#include <sys/socket.h>
+#include <unistd.h>
+#include <fstream>
+#include <netdb.h>
+#include <string>
+
+#include "CommunicationManager/ServerCommunicationManager.h"
+
+// When we are the responsibles
+
+AvailableConnection serverConfigFile[5];
+
+void loadServerConfigFile() {
+
+    // Mock connections
+    AvailableConnection fstConnection;
+    fstConnection.id = 219;
+    fstConnection.connectionInfo.ipAddress = "localhost";
+    fstConnection.connectionInfo.port = 2000;
+
+    AvailableConnection  sndConnection;
+    sndConnection.id = 12;
+    sndConnection.connectionInfo.ipAddress = "localhost";
+    sndConnection.connectionInfo.port = 2001;
+
+    AvailableConnection  thirdConnection;
+    thirdConnection.id = 45;
+    thirdConnection.connectionInfo.ipAddress = "localhost";
+    thirdConnection.connectionInfo.port = 2002;
+
+
+    AvailableConnection  fourthConnection;
+    fourthConnection.id = 6;
+    fourthConnection.connectionInfo.ipAddress = "localhost";
+    fourthConnection.connectionInfo.port = 2003;
+
+    AvailableConnection  fifthConnection;
+    fifthConnection.id = 100;
+    fifthConnection.connectionInfo.ipAddress = "localhost";
+    fifthConnection.connectionInfo.port = 2004;
+
+    serverConfigFile[0] = fstConnection;
+    serverConfigFile[1] = sndConnection;
+    serverConfigFile[2] = thirdConnection;
+    serverConfigFile[3] = fourthConnection;
+    serverConfigFile[4] = fifthConnection;
+}
+
+void ServerElectionManager::setupElectionManager(int serverID, ServerCommunicationManager* serverCommunicationManager) {
+    loadServerConfigFile();
+    this->setServerID(serverID);
+    this->setupCoordinator();
+    this->communicationManager = serverCommunicationManager;
+}
+
+void ServerElectionManager::setupElection() {
+    // Acessar minha lista de conexoes disponivel
+    // Se conectar com o socket no próximo
+    // Mandar uma mensagem de start election pro proximo server da lista junto com o ID
+
+
+    // Getting my index in the connections list using myID as reference
+    int myIndex = 0;
+
+    for (AvailableConnection &serverConnection: serverConfigFile) {
+        if (serverConnection.id == this->serverID) break;
+        myIndex++;
+    }
+
+    // Checking whats the server im about to connect
+
+    int numberOfConnections =  sizeof(serverConfigFile)/sizeof(AvailableConnection);
+
+    AvailableConnection nextConnetion;
+
+    int nextConnectionIndex =  ((numberOfConnections -1) == myIndex) ? 0 : (myIndex + 1);
+
+    if(serverConfigFile[nextConnectionIndex].id == this->elected) {
+        nextConnectionIndex =  ((numberOfConnections -1) == nextConnectionIndex) ? 0 : (nextConnectionIndex + 1);
+        nextConnetion = serverConfigFile[nextConnectionIndex];
+    } else {
+        nextConnetion = serverConfigFile[nextConnectionIndex];
+    }
+
+    this->elected = -1;
```

Comment:
Criar uma constante pro -1 que adicione valor semântico. Isso é `NONE_ELECTED`, `ELECTION_RUNNING` ou algo assim?
```suggestion
    this->elected = NONE_ELECTED;
```

Reply:
Boa

Reply:
Criei ELECTION_RUNNING